src/Sarif/Readers/DelegatingStream.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+
+namespace Microsoft.CodeAnalysis.Sarif
+{
+    internal class DelegatingStream : Stream
```

Comment:
>DelegatingStream [](http://example.com/codeflow?start=19&length=16)

Please go author this test, instantiate this type, dispose it and dispose it again.

Does this wrapper then raise an exception? i.e., are we expecting that multiple dispose methods are a no-op on the underlying stream? is it important for this class to allow one user to dispose it and then never again (despite never touching the wrapped object)?

Reply:
Just added a test with this behavior.
It will only thrown an exception if you call the `DisposeUnderlyingStream` and try to use the stream.